ruby/main.rb

```
@@ -359,4 +361,9 @@
 puts "Random number (1-100): #{generate_random_number(1, 100)}"
 puts
 
-puts "=== Ruby Demo Complete ===" 
\ No newline at end of file
+puts undefined_variable
+
+puts "=== Ruby Demo Complete ===" 
+
+if true
+  puts 'This will cause a syntax error' 
```

RuboCop (1.75.5)

[fatal] unexpected token $end
(Using Ruby 2.7 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

(Lint/Syntax)